src/prop/cnf_stream.cpp

```diff
@@ -293,14 +286,19 @@ SatLiteral CnfStream::convertAtom(TNode node)
 
 SatLiteral CnfStream::getLiteral(TNode node) {
   Assert(!node.isNull()) << "CnfStream: can't getLiteral() of null node";
-
   Assert(d_nodeToLiteralMap.contains(node))
-      << "Literal not in the CNF Cache: " << node << "\n";
+      << "CnfStream: literal for " << node << " not in the CNF Cache\n";
+
+  auto it = d_nodeToLiteralMap.find(node);
+  if (it != d_nodeToLiteralMap.end())
+  {
+    Trace("cnf") << "CnfStream::getLiteral(" << node << ") => " << it->second
+                << "\n";
+    return it->second;
+  }
 
-  SatLiteral literal = d_nodeToLiteralMap[node];
-  Trace("cnf") << "CnfStream::getLiteral(" << node << ") => " << literal
-               << "\n";
-  return literal;
+  Trace("cnf") << "CnfStream::getLiteral(" << node << ") failed!\n";
+  return undefSatLiteral;
```

Comment:
It should never be the case that getLiteral() is called for a node that does not have a literal. See the Assert in line 289. Maybe change the code that first calls find, asserts that it != d_nodeToLiteralMap.end() and return it->second.

Reply:
I see, the goal here was to optimize things by eliminating the need for the user to call both the find and get methods. This was especially meant for cases like https://github.com/cvc5/cvc5/blob/cf428345ee86032b5fccccea5806ecd6ae4cc623/src/prop/minisat/sat_proof_manager.cpp#L704-L721.

Reply:
Although, indeed, it could lead to undesirable breaking of the program if assertions are on. I'll leave the hypothesis of the assertion as it is and change the behavior of the method then.